frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -109,12 +110,51 @@ func Dockefile2Outline(ctx context.Context, dt []byte, opt ConvertOpt) (*outline
 	return &o, nil
 }
 
+func lintSourceInfo(srcMap *llb.SourceMap, srcRange *parser.Range) (filename string, source []string) {
+	if srcMap == nil {
+		return
+	}
+	filename = srcMap.Filename
+	if srcRange == nil {
+		return
+	}
+	lines := strings.Split(string(srcMap.Data), "\n")
+	start := srcRange.Start.Line
+	end := srcRange.End.Line
+	source = lines[start-1 : end]
+	return
+}
+
+func DockerfileLint(ctx context.Context, dt []byte, opt ConvertOpt) ([]linter.Warning, error) {
+	warnings := []linter.Warning{}
+	opt.Warn = func(short, url string, detail [][]byte, location *parser.Range) {
+		var source []string
+		var filename string
+		filename, source = lintSourceInfo(opt.SourceMap, location)
+		warnings = append(warnings, linter.Warning{
+			Short:    short,
+			Detail:   detail,
+			Location: location,
+			Filename: filename,
```

Comment:
We should try to keep the `SourceInfo` type in here. Maybe even list of warnings ans list of sourceinfo and then warning has a sourceinfo index. This is what we do in provenance for example for each build step.